src/prime_server.cpp

```diff
@@ -148,9 +148,10 @@ namespace prime_server {
 
   template <class request_container_t, class request_info_t>
   void server_t<request_container_t, request_info_t>::handle_timeouts() {
-    //kill timed out requests, NOTE: setting a huge request time out would be very stupid
+    //kill timed out requests
     auto drop_dead = static_cast<int64_t>(difftime(time(nullptr), 0) + .5) - request_timeout;
     while(drop_dead > 0 && request_history.size() && request_history.front().time_stamp < drop_dead) {
+      interrupt.send(static_cast<void*>(&request_history.front()), sizeof(uint64_t), ZMQ_DONTWAIT);
```

Comment:
In some places you've used `decltype(job)` rather than `uint64_t` directly. They're the same type, so it doesn't make much difference. I just wondered if it was important to be consistent.